lib/usecases/login_use_case.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/api/repository/authentication_repository.dart';
+import 'package:flutter_survey/usecases/base/base_use_case.dart';
+import 'package:injectable/injectable.dart';
+import 'package:flutter_survey/model/login_model.dart';
+
+class LoginInput {
+  final String email;
+  final String password;
+
+  LoginInput({
+    required this.email,
+    required this.password,
+  });
+}
+
+@Injectable()
+class LoginUseCase extends UseCase<LoginModel, LoginInput> {
```

Comment:
Optional: usecase should start with a verb too, while the model should be a noun

```suggestion
class LogInUseCase extends UseCase<LoginModel, LoginInput> {
```